### PR TITLE
Update more hooks and components to use trackRef

### PR DESCRIPTION
**Pull request by lukasIO**

### CODE REVIEW

**Comment by Ocupe** [comment]
Looks good so far. 👍 

General note: Most, if not all, deprecation messages and doc strings will show up on the documentation page. Therefore it would be great if they were formatted like a normal sentence (uppercase first letter and proper punctuation). Otherwise it may look odd and inconsistent.

File: packages/react/src/hooks/useTrackByName.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { useEnsureTrackRef } from '../context';
+import type { UseMediaTrackOptions } from './useMediaTrack';
+import { useMediaTrackBySourceOrName } from './useMediaTrackBySourceOrName';
+import type { TrackReferenceOrPlaceholder } from '@livekit/components-core';
+
+/**
+ * @public
+ * @deprecated this function will be removed in a future version, use `useTrackByName` instead
```
Comment (Ocupe) [inline packages/react/src/hooks/useTrackByName.ts:8]: Why would you want to add it and deprecate it in the same commit?

File: packages/react/src/hooks/useTrackByName.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ */
+export function useMediaTrackByName(
```
Comment (Ocupe) [inline packages/react/src/hooks/useTrackByName.ts:10]: File name and hook are named differently.

File: .changeset/sharp-ducks-whisper.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"@livekit/components-core": minor
+"@livekit/components-react": minor
+---
+
+Update more hooks and components to use `trackRef` and participant/source property pairs
```
Comment (Ocupe) [inline .changeset/sharp-ducks-whisper.md:6]: Did you mean "...instead of participant/source property pairs"?

**Review by Ocupe — APPROVED** [review]
Looks good, some small comments and doc comment suggestions along the way. 
We should also talk about how long the deprecation period will be.

File: packages/react/src/hooks/useIsMuted.ts [hunk 4]
```diff
@@ -9,14 +13,28 @@ export interface UseIsMutedOptions {
...
+  );
 
   React.useEffect(() => {
-    const listener = mutedObserver(p, source).subscribe(setIsMuted);
+    const listener = mutedObserver(ref).subscribe(setIsMuted);
     return () => listener.unsubscribe();
-  }, [p, source]);
+  }, [getTrackReferenceId(ref)]);
 
   return isMuted;
```
[the remark this answers is not in the document]
Comment (lukasIO) [reply packages/react/src/hooks/useIsMuted.ts:39]: I'm not sure I follow, you mean the `mutedObserver` gets it by source? everything is ref based here afaict

File: packages/react/src/hooks/useMediaTrackByName.ts [hunk 5]
```diff
@@ -3,7 +3,10 @@ import { useEnsureParticipant } from '../context';
 import type { UseMediaTrackOptions } from './useMediaTrack';
 import { useMediaTrackBySourceOrName } from './useMediaTrackBySourceOrName';
 
-/** @public */
+/**
+ * @public
+ * @deprecated this function will be removed in a future version, use `useTrackByName` instead
```
[the remark this answers is not in the document]
Comment (lukasIO) [reply packages/react/src/hooks/useMediaTrackByName.ts:8]: simply forgot to commit this 🤦

File: packages/react/src/hooks/useTrackByName.ts (hunk 1 above)
Comment (lukasIO) [reply packages/react/src/hooks/useTrackByName.ts:8]: yeah, that was copy+paste...

File: .changeset/sharp-ducks-whisper.md (hunk 3 above)
Comment (lukasIO) [reply .changeset/sharp-ducks-whisper.md:6]: "deprecate"

File: packages/react/src/hooks/useTrack.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import { useMediaTrackBySourceOrName } from './useMediaTrackBySourceOrName';
+
+/** @public */
+export interface UseTrackOptions {
+  element?: React.RefObject<HTMLMediaElement>;
+  props?: React.HTMLAttributes<HTMLVideoElement | HTMLAudioElement>;
+}
+
+/** @public */
+export function useTrack(trackRef: TrackIdentifier, options: UseTrackOptions = {}) {
```
[the remark this answers is not in the document]
Comment (lukasIO) [reply packages/react/src/hooks/useTrack.ts:12]: yeah, that's a good point, but I dread the sound of it 😅

**Pull request merged**